system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -1121,30 +1121,40 @@ class CLdapClient : public CInterface, implements ILdapClient
     virtual void init(IPermissionProcessor* pp)
     {
         m_pp = pp;
-        if(m_ldapconfig->getServerType() == OPEN_LDAP)
+        static CriticalSection  lcCrit;
+        static bool createdOU = false;
+        if (!createdOU)
         {
-            try
-            {
-                addDC(m_ldapconfig->getBasedn());
-            }
-            catch(...)
-            {
-            }
-            try
-            {
-                addGroup("Directory Administrators", m_ldapconfig->getBasedn());
-            }
-            catch(...)
+            CriticalBlock block(lcCrit);
+            if (!createdOU)
```

Comment:
I think what Gavin means is that as far as YOU are concerned, it needs to be re-evaluated... but as far as the compiler is concerned, it doesn't... so the compiler may decide not to bother.